app/Http/Controllers/App/AuthController.php

```diff
@@ -54,13 +58,50 @@ public function login(Request $request)
      */
     public function logout(Request $request)
     {
-        if (!Auth::check()) {
-            return self::json([], 401, ['message' => 'Not Authorized']);
-        }
-
         Auth::guard()->logout();
         $request->session()->invalidate();
 
-        return self::json([], 200);
+        return self::json([], 204);
+    }
+
+    /**
+     * Log the user out of the application.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function recovery(Request $request)
+    {
+        Validator::make($request->all(), ['email' => 'required|email', 'uri' => 'required'])->validate();
+        $user = User::where('email', $request->input('email'))->first();
+        if (empty($user)) {
+            return self::json([], 204);
+        }
+        DB::beginTransaction();
+        if (!Token::canGenerate($user->id, 'password-recovery', $this->password_recovery_resend_limit)) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "password-recovery" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)